cmd/malwareExecuteScan.go

```diff
@@ -118,12 +87,55 @@ func runMalwareScan(config *malwareExecuteScanOptions, telemetryData *telemetry.
 	if scanResponse.MalwareDetected || scanResponse.EncryptedContentDetected {
 		return fmt.Errorf("Malware scan failed for file '%s'. Malware detected: %t, encrypted content detected: %t",
 			config.File, scanResponse.MalwareDetected, scanResponse.EncryptedContentDetected)
-	} else {
-		log.Entry().Infof("Malware scan succeeded for file '%s'. Malware detected: %t, encrypted content detected: %t",
-			config.File, scanResponse.MalwareDetected, scanResponse.EncryptedContentDetected)
+	} 
+	
+	log.Entry().Infof("Malware scan succeeded for file '%s'. Malware detected: %t, encrypted content detected: %t",
+		config.File, scanResponse.MalwareDetected, scanResponse.EncryptedContentDetected)
+
+	return nil
+}
+
+func sendMalwareScanRequest(client piperhttp.Sender, method, url string, candidate io.Reader, header http.Header, cookies []*http.Cookie) (*malwareExecuteScanResponse, error) {
+
+	// piper http utils mashall some http response codes into errors. We wan't to check the status code
+	// ourselvs hence we wait with returning that error (maybe also related to errors others than http status codes)
+
+	response, err := client.SendRequest(method, url, candidate, header, cookies)
+
+	return validateResponse( response, err )
+}
+
+func validateResponse(response *http.Response, err error) (*malwareExecuteScanResponse, error) {
+	
+	var body []byte = []byte("n/a")
+	if response != nil && response.Body != nil {
+		body, _ = ioutil.ReadAll(response.Body)
+		response.Body.Close()
+	}
+
+	if(err != nil) {
+		return nil, fmt.Errorf("HTTP request failed with error: %v. Details: \"%s\"", err, body)
 	}
 
-	return err
+	if response.StatusCode != 200 {
+		return nil, fmt.Errorf("Unexpected response code (%d). %d expected. Details: \"%s\"", response.StatusCode, 200, body)
+	}
+
+	return scanResponseBody(body)
```

Comment:
The unlikely - but not impossible case - of `response.Status == 200` and an empty or nil response body would end up in a failure parsing the `<n/a>`.

Reply:
ok. what should happen in case of an empty body?

furthermore, what about 2XX and 3xx response codes?

Reply:
> ok. what should happen in case of an empty body?

In that case we should fail since the repsonse is not as expected. We expect a respone like

```
{"malwareDetected":false,"encryptedContentDetected":false,"scanSize":"999", mimeType":"myType","SHA256":"<hash>"}
```

> furthermore, what about 2XX and 3xx response codes?

To my understanding `200` is the only valid response code in the `2xx` range.
Regarding the `3xx` range I expect the same behaviour like `curl --location`, but I did not test that (... would be hard to get a reality check for that).